src/dotnet/Users.UI.Blazor/Components/OwnAvatarEditorModal.razor

```diff
@@ -158,6 +158,14 @@
     private void OnCancel()
         => CloseModal();
 
+    private async Task OnDeleteAvatarClick() {
+        var command = new IAvatars.ChangeCommand(Session, ModalModel.AvatarId, null, new() {
```

Comment:
May be it make sense to warn user that the avatar is used in one or more chats?

Reply:
It's kinda hard to do now - i.e. only in a bit hacky way (e.g. make IChats to provide this info by querying all the chats). We'll need to store the list of chats against avatars in future.

Reply:
But overall, I agree, something like this is necessary here.

Reply:
(maybe as a separate task)

Reply:
What we can do, actually, is to mark the avatar as "removed" & make it still available in chats. I.e. an easy & kinda non-confusing solution. Avatar picker may show such avatar for chats as well (just a single one, if it's the current one), marking it as "Removed".